From 670dd437ce656503290bf40d124013421ae72873 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kamil=20Go=C5=82da?= <kamil.golda@protonmail.com>
Date: Wed, 2 Oct 2024 12:00:17 +0200
Subject: [PATCH 16/20] arm64: dts: qcom: sm6125-xiaomi-laurel-sprout: Add USB
 OTG support and pmi632 type-c node
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Set dr_mode to otg, add usb-role-switch property, and a port
in usb3_dwc3 in sm6125.dtsi
Connect the port, add pmi632_typec and pmi632_vbus nodes in laurel dts.

Signed-off-by: Kamil Gołda <kamil.golda@protonmail.com>
[rebase]
Signed-off-by: Barnabás Czémán <barnabas.czeman@mainlining.org>
---
 .../dts/qcom/sm6125-xiaomi-laurel-sprout.dts  | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts b/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
index 139f2b4..944c7ae 100644
--- a/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
+++ b/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
@@ -11,6 +11,7 @@
 #include <dt-bindings/pinctrl/qcom,pmic-gpio.h>
 #include "sm6125.dtsi"
 #include "pm6125.dtsi"
+#include "pmi632.dtsi"
 
 / {
 	model = "Xiaomi Mi A3";
@@ -311,6 +312,33 @@ vol_up_n: vol-up-n-state {
 	};
 };
 
+&pmi632_typec {
+	status = "okay";
+
+	connector {
+		compatible = "usb-c-connector";
+
+		power-role = "dual";
+		data-role = "dual";
+		self-powered;
+
+		typec-power-opmode = "default";
+		pd-disable;
+
+		port {
+			pmi632_hs_in: endpoint {
+				remote-endpoint = <&usb_dwc3_hs>;
+			};
+		};
+	};
+};
+
+&pmi632_vbus {
+	regulator-min-microamp = <500000>;
+	regulator-max-microamp = <1000000>;
+	status = "okay";
+};
+
 &pon_pwrkey {
 	status = "okay";
 };
@@ -578,3 +606,7 @@ &usb3 {
 &usb3_dwc3 {
 	extcon = <&extcon_usb>;
 };
+
+&usb_dwc3_hs {
+	remote-endpoint = <&pmi632_hs_in>;
+};
-- 
2.56.0

